### ISSUE TITLE
Angular proxy fails when using Chrome devtools on the MacBook Pro

### ISSUE THREAD

**Issue by jimrand1** [angular/angular-cli#28394]
### Is this a bug report or a feature request?

- [X] Bug Report
- [ ] Feature Request

### Please provide the steps to reproduce the issue [Bug Report only]

Steps to reproduce the problem

Create a simple Express server to start on port 3000
Create a simple Angular app to start on port 4200
Create a proxy.conf.json file containing: { "/api/*": { "target": "http://localhost:3000/", "changeOrigin": true, "logLevel": "debug", "secure": false } }
In Angular, call the API: await lastValueFrom(this.httpClient.get(/api/quotes, { params: uriParams}))
Problem Description

When not using devtools, the api call works. When in the debugger, the call fails to http://localhost:4200/api/quotes. It should go to port 3000. Rebuilding both projects in Linux (Ubuntu), results with successful API calls - both without the Chrome devtools and with Chrome devtools. Maybe this is a simple configuration issue. Impact right now is Angular development limited to no client side debugging on the Mac.

Summary

Angular proxy fails when using Chrome devtools on the MacBook Pro[
[DevTools Test.docx](https://github.com/user-attachments/files/16964233/DevTools.Test.docx)
[DevTools Test.docx](https://github.com/user-attachments/files/16964237/DevTools.Test.docx)
](url)

### Is this a browser-specific issue? If so, please specify the device, browser, and version. [Bug Report only]

```true
This is a Chrome issue which they wanted moved over here
https://issues.chromium.org/issues/364904750
```

**Comment by alan-agius4** [comment]
Can you please provide the output of `ng version`?

**Comment by jimrand1** [comment]
Angular CLI: 18.2.3
Node: 20.16.0
Package Manager: npm 10.8.1
OS: darwin arm64

Angular: 18.2.3
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1802.3
@angular-devkit/build-angular   18.2.3
@angular-devkit/core            18.2.3
@angular-devkit/schematics      18.2.3
@schematics/angular             18.2.3
rxjs                            7.8.1
typescript                      5.5.4
zone.js                         0.14.10

**Comment by jimrand1** [comment]
By the way, the error is "no internet connection" when running with developer tools open.

**Comment by JoostK** [comment]
Does this repro when using `withFetch` for Angular's HTTP client? Alternatively, does this repro when using native `fetch` instead of HttpClient?

Additionally, could you please provide a runnable reproduction instead of individual steps to get there?

**Comment by jimrand1** [comment]
Hi Joost,

See attached zip file

Jim


On Sep 12, 2024, at 7:38 AM, Joost Koehoorn ***@***.***> wrote:



Does this repro when using withFetch for Angular's HTTP client? Alternatively, does this repro when using native fetch instead of HttpClient?

Additionally, could you please provide a runnable reproduction instead of individual steps to get there?

**Comment by jimrand1** [comment]
Zip file here - two projects - client and server.  Thank you for looking into this. 
[ProxyFail.zip](https://github.com/user-attachments/files/16980268/ProxyFail.zip)

**Comment by jimrand1** [comment]
Just uploaded file to GitHub.

Jim


On Sep 12, 2024, at 9:37 AM, Alan Agius ***@***.***> wrote:



@jimrand1<https://github.com/jimrand1>, looks like the zip file was not uploaded.

**Comment by alan-agius4** [comment]
@jimrand1, I am not able to replicate the problem you are experiencing even when using Mac.
<img width="1403" alt="Screenshot 2024-09-12 at 16 05 04" src="https://github.com/user-attachments/assets/87d6f637-f20c-4f39-94d7-2ac526a1e853">

**Comment by alan-agius4** [comment]
@jimrand1, yes please see screenshot above.

**Comment by JoostK** [comment]
I also cannot reproduce, using Chrome 128.0.6613.85 on macOS, it works regardless of the Web Inspector being open.

**Comment by jimrand1** [comment]
The screen shot you provided does not show Chrome developer tools opened. Attached are two images. The first is before opening the developer tools. The second is after opening the developer tools and refreshing the browser.
<img width="811" alt="Screenshot 2024-09-12 at 12 50 30 PM" src="https://github.com/user-attachments/assets/96bc8bf8-9628-47fb-bdf1-91ab91e1b153">
<img width="880" alt="Screenshot 2024-09-12 at 12 51 30 PM" src="https://github.com/user-attachments/assets/45b96a17-9a7e-4a98-945b-d904e1a67d10">

**Comment by JoostK** [comment]
The screenshot does contain Devtools, it is the dark mode bit on the left.

But I notice something in your screenshot, which is that the network tab shows a warning symbol. Which indicates to me that you have "Throttling" enabled and set to "Offline", which would explain _everything_.

**Comment by alan-agius4** [comment]
Closing as mention by @JoostK, the “Throttling” is set to “offline”.

**Comment by jimrand1** [comment]
Hi Alan,

Thank you. You nailed it.  Throttling was the problem.

Thanks again.

Jim


On Sep 12, 2024, at 1:33 PM, Alan Agius ***@***.***> wrote:



Closed #28394<https://github.com/angular/angular-cli/issues/28394> as not planned.

**Comment by alan-agius4** [comment]
It was actually @JoostK you noticed the problem 😀
